sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py

```diff
@@ -338,30 +345,35 @@ def begin_update(
                 feature_store.name,
                 rest_workspace_obj.feature_store_settings.online_store_connection_name,
             )
+
             online_store_target_to_update = (
                 online_store_target_to_update or existing_online_store_connection.properties.target
             )
             if online_store and (
                 not existing_online_store_connection.properties
                 or existing_online_store_connection.properties.target != online_store.target
             ):
+                update_online_store_connection = True
                 update_online_store_role_assignment = True
                 module_logger.info(
                     "Warning: You have changed the online store connection, "
                     "any data that was materialized earlier "
                     "will not be available. You have to run backfill again."
                 )
         elif online_store_target_to_update:
+            update_online_store_connection = True
             update_online_store_role_assignment = True
 
         feature_store_settings = FeatureStoreSettings._from_rest_object(rest_workspace_obj.feature_store_settings)
 
+        # generate a random suffix for online/offline store connection name
+        random_string = uuid.uuid4().hex[:8]
         if offline_store:
             if materialization_identity:
                 offline_store_connection_name = (
                     feature_store_settings.offline_store_connection_name
-                    if feature_store_settings.offline_store_connection_name
-                    else OFFLINE_STORE_CONNECTION_NAME
+                    if not update_offline_store_connection
+                    else f"{OFFLINE_STORE_CONNECTION_NAME}-{random_string}"
                 )
                 offline_store_connection = WorkspaceConnection(
```

Comment:
only create workspace connection when need to update store connection?

Reply:
Yes. if update_offline_store_connection/update_online_store_connection is false, means the target is not changed, then it will keep the existing workspace connection.

Reply:
I mean we don't need to call backend to create workspace connection when update_offline_store_connection/update_online_store_connection is false

Reply:
Skipped the connection creation when the update_offline_store_connection/update_online_store_connection is false.